c7n/resources/s3.py

```diff
@@ -789,6 +1029,212 @@ def process_bucket(self, func, bucket, topic, account_id, session_factory):
         return source.add(func)
 
 
+@actions.register('configure-lifecycle')
+class ConfigureLifecycle(BucketActionBase):
+    """Action that aggregates all rules with whole bucket prefix and
+    changes them to the configured policy values."""
+
+    permissions = ('s3:GetLifecycleConfiguration', 's3:PutLifecycleConfiguration', )
+
+    schema = {
+        'type': 'object',
+        'additionalProperties': True,
+        'properties': {
+            'type': {'enum': ['configure-lifecycle']},
+            'id': {'type': 'string'},
+            'prefix': {'type': 'string'},
+            'deletes_objects': {'type': 'boolean'},
+            'delete_days': {'type': 'number'},
+            'multipart_days': {'type': 'number'},
+            'ia_days': {'type': 'number'},
+            'glacier_days': {'type': 'number'},
+            'max_workers': {'type': 'number'}
+        }
+    }
+
+    cache_update = False
+
+    def process(self, buckets):
+            max_workers = self.data.get('max_workers', DEFAULT_WORKERS)
+
+            with self.executor_factory(max_workers=max_workers) as w:
+                saved_buckets = w.map(self.call_api, buckets)
+                saved_buckets = filter(None, list(saved_buckets))
+
+            if self.cache_update:
+                saved_buckets = {b['Name']: b for b in saved_buckets}
+                self.manager._cache.save(CACHE_NAME, saved_buckets)
+
+            with self.executor_factory(max_workers=max_workers) as w:
+                w.map(self.put_lifecycle, buckets)
+
+    def put_lifecycle(self, bucket):
+        prefix = self.data.get('prefix', '')
+
+        # Make sure lifecycle is within configured policy limits
+        ia_days_in_policy = self.data.get('ia_days', None)
+        glacier_days_in_policy = self.data.get('glacier_days', None)
+        delete_days_in_policy = self.data.get('delete_days')
+        multipart_days_in_policy = self.data.get('multipart_days', None)
+
+        session = local_session(self.manager.session_factory)
+        s3 = bucket_client(session, bucket)
+        bname = bucket['Name']
+
+        # Glacier storage is not enabled in sa-east-1 or ap-northeast-1
+        bucket_region = get_bucket_region(bucket)
+        glacier_supported_regions = session.get_available_regions('glacier')
+
+        rem_lifecycle = S3BucketLifecycle(prefix, self.data.get('id'))
+        if ia_days_in_policy:
+            rem_lifecycle.set_ia_transition_days(ia_days_in_policy)
+        if glacier_days_in_policy and (bucket_region in glacier_supported_regions):
+            rem_lifecycle.set_glacier_transition_days(glacier_days_in_policy)
+        if multipart_days_in_policy:
+            rem_lifecycle.set_stale_multipart_uploads_days(multipart_days_in_policy)
+        if delete_days_in_policy:
+            rem_lifecycle.set_deletes_objects_days(delete_days_in_policy)
+
+        bad_bucket_lifecycle = True
```

Comment:
I feel like this code is more intuitive if this is called `changed_bucket_lifecycle` instead of `bad_bucket_lifecycle`, since this boolean is not being used to track whether the lifecycle is invalid (like what is happening in `has-lifecycle` above) but rather it is being used to track whether the lifecycle specified is either new or it changes an existing one